exporters/otlp/profiles/src/main/java/io/opentelemetry/exporter/otlp/profiles/LocationMarshaler.java

```diff
@@ -8,29 +8,28 @@
 import io.opentelemetry.exporter.internal.marshal.MarshalerUtil;
 import io.opentelemetry.exporter.internal.marshal.MarshalerWithSize;
 import io.opentelemetry.exporter.internal.marshal.Serializer;
-import io.opentelemetry.proto.profiles.v1experimental.internal.Location;
+import io.opentelemetry.proto.profiles.v1development.internal.Location;
 import java.io.IOException;
 import java.util.List;
 import java.util.function.Consumer;
+import javax.annotation.Nullable;
 
 final class LocationMarshaler extends MarshalerWithSize {
 
   private static final LocationMarshaler[] EMPTY_REPEATED = new LocationMarshaler[0];
 
-  private final long mappingIndex;
+  @Nullable private final Integer mappingIndex;
```

Comment:
Nit: Does it make sense to use `java.util.Optional` or does that just add headache here?

Reply:
We don't use `Optional` anywhere in this repository. The logic goes something like this:
- For external user facing APIs, optional creates a tri-state, with internals needing to check for null, present, and empty, since even with assuming non-null arguments and `@Nullable` annotations where arguments are nullable, we often perform additional null checks.
- For internal APIs, optinoal adds additional allocation vs. `@Nullable`, and maintainers can / should accept higher cognitive load in exchange for better performance.

Reply:
(https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/style-guideline.md#javautiloptional-usage)